src/sieve.py

```diff
@@ -15,6 +15,11 @@ def sieve(n: int) -> list[int]:
     candidates = list(range(2, n + 1))
     primes = []
 
-    # FIXME: fill out this bit
-
+    while len(candidates) > 0:
```

Comment:
You can just write `while candidates` here. An empty list evaluates to `False` while a non-empty list evaluates to `True`.